File: native/android/app/src/main/java/org/membercovenantpath/viewer/ui/components/OrgFilterBar.kt
package org.membercovenantpath.viewer.ui.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.FilterAltOff
import androidx.compose.material3.AssistChip
import androidx.compose.material3.AssistChipDefaults
import androidx.compose.material3.FilterChip
import androidx.compose.material3.FilterChipDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import org.membercovenantpath.viewer.logic.OrgBucket
import org.membercovenantpath.viewer.logic.Orgs

/**
 * Org-ownership filter bar: a colored toggle chip per org (WML / EQ / RS), ALL selected by default.
 * Tapping toggles (multi-select; can't deselect the last). When not all three are on, a "Clear
 * filters" affordance restores them. Colors come from [Orgs.info] so they stay consistent with the
 * per-member responsibility chips. Ported from golden_hour_view.dart `_OrgFilterBar`.
 */
@Composable
fun OrgFilterBar(
    selected: Set<OrgBucket>,
    onToggle: (OrgBucket) -> Unit,
    onClear: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val all = selected.size == OrgBucket.entries.size
    FlowLayout(modifier = modifier, horizontalSpacing = 6.dp, verticalSpacing = 6.dp) {
        OrgBucket.entries.forEach { b ->
            val info = Orgs.info(b)
            val sel = b in selected
            FilterChip(
                selected = sel,
                onClick = { onToggle(b) },
                label = {
                    Text(
                        info.label,
                        color = if (sel) info.color else info.color.copy(alpha = 0.6f),
                        fontWeight = if (sel) FontWeight.Bold else FontWeight.Medium,
                    )
                },
                leadingIcon = {
                    Icon(info.icon, contentDescription = null, tint = info.color, modifier = Modifier.size(16.dp))
                },
                colors = FilterChipDefaults.filterChipColors(
                    containerColor = info.color.copy(alpha = 0.06f),
                    selectedContainerColor = info.color.copy(alpha = 0.22f),
                ),
                border = BorderStroke(1.dp, info.color.copy(alpha = if (sel) 0.9f else 0.35f)),
            )
        }
        if (!all) {
            AssistChip(
                onClick = onClear,
                label = { Text("Clear filters") },
                leadingIcon = {
                    Icon(Icons.Outlined.FilterAltOff, contentDescription = null, modifier = Modifier.size(16.dp))
                },
            )
        }
    }
}

File: native/ios/Sources/CovenantPathKit/Views/Shared/OrgFilterBar.swift
#if canImport(UIKit)
import SwiftUI

/// The org-ownership filter bar (Golden Hour + Needs): a colored toggle per org (WML / EQ / RS), ALL
/// selected by default. Tapping a chip toggles whether that org's converts are shown (multi-select,
/// additive); you can't deselect the last one (→ empty). When not all three are on, a "Clear filters"
/// chip restores them. Colors come from `Org.info` so they stay consistent with the per-member
/// responsibility chips. Faithful port of `_OrgFilterBar` + the `_toggleOrg` guard.
public struct OrgFilterBar: View {
    @Binding var selected: Set<OrgBucket>

    public init(selected: Binding<Set<OrgBucket>>) { self._selected = selected }

    private var allSelected: Bool { selected.count == OrgBucket.allCases.count }

    /// Toggle that never lets the user deselect the last org (delegates to the pure `Org.toggleFilter`
    /// so the rule has one home and stays unit-testable without UIKit).
    public static func toggle(_ b: OrgBucket, in set: inout Set<OrgBucket>) {
        Org.toggleFilter(b, in: &set)
    }

    public var body: some View {
        // Liquid-Glass capsules (frosted-material fallback on iOS 17–25). The glass container lets the
        // adjacent chips blend fluidly on iOS 26.
        CPGlassGroup(spacing: 6) {
            FlowLayout(spacing: 6) {
                ForEach(OrgBucket.allCases) { b in
                    let info = Org.info(b)
                    let color = Color(hex: info.hex)
                    let sel = selected.contains(b)
                    Button {
                        OrgFilterBar.toggle(b, in: &selected)
                    } label: {
                        HStack(spacing: 6) {
                            Image(systemName: info.symbol).font(.system(size: 14))
                                .foregroundStyle(color)
                            Text(info.label)
                                .font(.system(size: 13, weight: sel ? .bold : .medium))
                                .foregroundStyle(sel ? color : color.opacity(0.65))
                        }
                        .padding(.horizontal, 13).padding(.vertical, 8)
                        .cpGlassChip(tint: color, selected: sel)
                    }
                    .buttonStyle(.plain)
                }
                if !allSelected {
                    Button {
                        selected = Set(OrgBucket.allCases)
                    } label: {
                        HStack(spacing: 6) {
                            Image(systemName: "line.3.horizontal.decrease.circle").font(.system(size: 14))
                            Text("Clear filters").font(.system(size: 13))
                        }
                        .padding(.horizontal, 13).padding(.vertical, 8)
                        .cpGlassChip(tint: Color(.secondaryLabel), selected: false)
                    }
                    .buttonStyle(.plain)
                    .foregroundStyle(.secondary)
                }
            }
        }
        .frame(maxWidth: .infinity)
    }
}

/// A centered, muted one-liner (the org-responsibility explanation under the filter). Mirrors
/// `_SubtleNote`.
public struct SubtleNote: View {
    let text: String
    public init(_ text: String) { self.text = text }
    public var body: some View {
        Text(text)
            .font(.caption)
            .foregroundStyle(.secondary)
            .multilineTextAlignment(.center)
            .frame(maxWidth: .infinity)
            .padding(.horizontal, 20)
    }
}
#endif
